.github/workflows/unit-test.yml

```diff
@@ -2,12 +2,19 @@ name: Unit Tests
 on: [workflow_dispatch, workflow_call]
 jobs:
   unit-test:
-    runs-on: ubuntu-latest
+    runs-on:
+      - runs-on
+      - spot=false
+      - runner=8cpu-linux-arm64
+      - run-id=${{ github.run_id }}
     timeout-minutes: 60
     steps:
       - name: Checkout code
-        uses: actions/checkout@08eba0b27e820071cde6df949e0beb9ba4906955 # v4
-
+        uses: actions/checkout@v4
```

Comment:
We've switched to pinned dependencies per https://github.com/rancher/rancher/pull/51593 so please revert this change.

Can we also pin `actions/setup-go` to whatever commit `v5` is on now, please?